backend/api_calls.py

```diff
@@ -1055,6 +1056,29 @@ def MarkNotificationSeen(request, firebase):
       request['playerId'], request['notificationId']), put_data)
 
 
+def RegisterUserDevice(request, firebase):
+  """Register a user device to a userId.
+
+  Validation:
+    userId must exist.
+
+  Args:
+    userId: User id to associate with.
+    deviceToken: Ionic device token.
```

Comment:
The main reason is that not everyone is likely to call /register from their device. They may do it from the website and then install the app. I think this should be separate and should be called everytime someone logs into an app via mobile. Whether this is part of the larger /updateUser call or specifically with a /registerUserDevice call. In this case, I would argue that a call specifically for this makes more sense given that we are specifically interested in registering a device for the user and not necessarily updating any other user related metadata.